tasks/apt.yml

```diff
@@ -0,0 +1,37 @@
+# SPDX-FileCopyrightText: 2025 Stefan Haun <tux@netz39.de>
+# SPDX-License-Identifier: MIT
+---
+
+- name: Install Docker APT deps
+  ansible.builtin.package:
+    name: "{{ packages }}"
+    state: present
+  vars:
+    packages:
+      - apt-transport-https
+      - ca-certificates
+      - gnupg2
+
+# Use import_tasks to gain the ability to register changed status for apt_repo
+
+- name: Include tasks for setting up APT key
+  ansible.builtin.import_tasks: apt_key.yml
+
+# TODO Determine the best way to detect if we need to use the legacy setup
```

Comment:
@LeSpocky Shall we fix this aspect in this PR or do the re-structuring first and then find a good way to determine which method to use?